core/src/main/java/com/yahoo/ycsb/TimeseriesDB.java

```diff
@@ -0,0 +1,282 @@
+   * @param value     actual value to insert
+   * @param tags      A HashMap of tag/tagvalue pairs to insert as tagsmv c
+   * @return A {@link Status} detailing the outcome of the insert
+   */
+  protected abstract Status insert(String metric, Long timestamp, double value, Map<String, ByteIterator> tags);
+
+  /**
+   * NOTE: This operation is usually <b>not</b> supported for Time-Series databases.
+   * Deletion of data is often instead regulated through automatic cleanup and "retention policies" or similar.
+   *
+   * @return Status.NOT_IMPLEMENTED or a {@link Status} specifying the outcome of deletion
+   * in case the operation is supported.
+   */
+  public Status delete(String table, String key) {
+    return Status.NOT_IMPLEMENTED;
+  }
+
+  protected final String buildTagFilter(Map<String, List<String>> tags) {
+    StringBuilder tagFilter = new StringBuilder();
+    for (String tag : tags.keySet()) {
+      tagFilter.append(" AND ( ");
+      tagFilter.append(tags.get(tag).stream()
+          .map(t -> String.format("%s = '%s'", tag, t))
+          .collect(Collectors.joining(" OR ")));
+      tagFilter.append(" )");
+    }
+    return tagFilter.toString();
+  }
+
+  /**
+   * Examines the given {@link Properties} and returns an array containing the Tag Keys
+   * (basically matching column names for traditional Relational DBs) that are detailed in the workload specification.
+   * See {@link TimeSeriesWorkload} for how these are generated.
+   * <p>
+   * This method is intended to be called during the initialization phase to create a table schema
+   * for DBMS that require such a schema before values can be inserted (or queried)
+   *
+   * @param properties
+   *      The properties detailing the workload configuration.
+   * @return An array of strings specifying all allowed TagKeys (or column names)
+   *      except for the "value" and the "timestamp" column name.
+   * @implSpec WARNING this method must exactly match how tagKeys are generated by the {@link TimeSeriesWorkload},
+   *    otherwise databases requiring this information will most likely break!
+   */
+  protected static String[] getPossibleTagKeys(Properties properties) {
+    final int tagCount = Integer.parseInt(properties.getProperty(TimeSeriesWorkload.TAG_COUNT_PROPERTY,
+        TimeSeriesWorkload.TAG_COUNT_PROPERTY_DEFAULT));
+    final int tagKeylength = Integer.parseInt(properties.getProperty(TimeSeriesWorkload.TAG_KEY_LENGTH_PROPERTY,
+        TimeSeriesWorkload.TAG_KEY_LENGTH_PROPERTY_DEFAULT));
+
+    Generator<String> tagKeyGenerator = new IncrementingPrintableStringGenerator(tagKeylength);
+    String[] tagNames = new String[tagCount];
+    for (int i = 0; i < tagCount; i++) {
+      tagNames[i] = tagKeyGenerator.nextValue();
+    }
+    return tagNames;
+  }
+
+
+  /**
+   * An enum containing the possible aggregation operations.
+   */
+  public enum AggregationOperation {
+    NONE, SUM, AVERAGE, COUNT;
```

Comment:
Would be good to have `MAX` and `MIN` as well, those are fairly common. And javadocs would be nice :)